metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestRetentionPoliciesStep.java

```diff
@@ -45,6 +50,10 @@ public String name() {
   @Override
   public void execute() throws IOException, URISyntaxException {
     // 0. Execute preflight check to see whether we need to ingest policies
+    if (_entityService.exists(UPGRADE_ID_URN)) {
```

Comment:
If we apply retention policies once, then change the retention values, with this static check won't it mean the new retention values will be skipped?

Reply:
There are 2 things 
- Setting retention policy
- Applying retention on URNs

This Step only adds default policies. We should not be overwriting them again every time. Retention is applied when ingestion happens which is not part of this. That is present in RetentionService which is called from various ingestion paths separately.

Reply:
There isn't a case for when we want to update the default retention policies?

Reply:
Through ingesting metadata. e.g. by sending this they can change the retention to be 5 iversions nstead of default of 20.
```
[
  {
    "auditHeader":null,
    "entityType":"dataHubRetention",
    "entityUrn": "urn:li:dataHubRetention:(*,*)",
    "changeType":"UPSERT",
    "aspectName":"dataHubRetentionConfig",
    "aspect":{
        "value": "{\"retention\":{\"version\":{\"maxVersions\":5}}}",
      "contentType":"application/json"
    },
    "systemMetadata":null
  }
]
```

This model is documented here https://datahubproject.io/docs/generated/metamodel/entities/datahubretention/#datahubretentionconfig

Reply:
@jjoyce0510 @anshbansal 
Does this not directly contradict what the docs say? ... 
```As of now, retention policies are applied in two places:

GMS boot-up: A bootstrap step ingests the predefined set of retention policies. If no policy existed before or the existing policy was updated, an asynchronous call will be triggered. It will apply the retention policy (or policies) to all records in the database.
Ingest: On every ingest, if an existing aspect got updated, it applies the retention policy to the urn-aspect pair being ingested.```

I am working through applying a new retention policy on my GMS k8 Pod and based on the documentation this has been my approach...
1. Mount a volume to pod at the path /etc/datahub/plugins/retention/retention.yaml
2. Confirm that this path holds a file with our retention policy yaml specified
3. Set the ENTITY_SERVICE_ENABLE_RETENTION : true
4. Restart the GMS server
5. We expect that the code would get executed to apply this policy to the DB as mentioned in the Docs

What we actually see is this output, confirming that this code is skipping the retention application on the data...
" Retention was applied. Skipping. " from the line added in this PR. We are in this position as our DataBase Aspect and Entity table have grown far too large ~20mill Aspects, and we want this policy to be applied to all aspects and enities.

Ex. of policy (in QA so we are being aggressive)
- entity: "*"
      aspect: "*"
      config:
          retention:
          version:
              maxVersions: 5

Reply:
It all makes sense up to the point of how to run this new .yaml file defining a DataHubRetention and its corresponding dataHubRetentionKey and dataHubRetentionConfig. Since the key is literally what aspects/entities are being set, and we are using the same key I assume the default policy uses. Then we are updating the default policy either way. We are having trouble following the logic above...
```[
  {
    "auditHeader":null,
    "entityType":"dataHubRetention",
    "entityUrn": "urn:li:dataHubRetention:(*,*)",
    "changeType":"UPSERT",
    "aspectName":"dataHubRetentionConfig",
    "aspect":{
        "value": "{\"retention\":{\"version\":{\"maxVersions\":5}}}",
      "contentType":"application/json"
    },
    "systemMetadata":null
  }
]```
even with this we are effectively updating the default policy.  Now how do we get it to apply to all entities and aspects, we don't run ingestions for a lot of them anymore, it would make more for this to happen view the async gms process.